include/ableton/Link.hpp

```diff
@@ -359,21 +359,31 @@ class Link
       bool isPlaying, std::chrono::microseconds time, double beat, double quantum);
 
   private:
-    friend Link;
+    friend BasicLink<Clock>;
     link::ApiState mOriginalState;
     link::ApiState mState;
     bool mbRespectQuantum;
   };
 
 private:
+  using Controller = ableton::link::Controller<
+    link::PeerCountCallback,
+    link::TempoCallback,
+    link::StartStopStateCallback,
+    Clock,
+    link::platform::Random,
+    link::platform::IoContext>;
+
   std::mutex mCallbackMutex;
-  link::PeerCountCallback mPeerCountCallback;
-  link::TempoCallback mTempoCallback;
-  link::StartStopStateCallback mStartStopCallback;
+  link::PeerCountCallback mPeerCountCallback = [](std::size_t) {};
+  link::TempoCallback mTempoCallback = [](link::Tempo) {};
+  link::StartStopStateCallback mStartStopCallback = [](bool) {};
   Clock mClock;
-  link::platform::Controller mController;
+  Controller mController;
 };
 
+using Link = BasicLink<link::platform::Clock>;
```

Comment:
In our setup we forward declare `Link`. This turned out to be complicated using a type alias. Therefore I suggest sticking with a `Link` class, as it is more in line with what we had before:
```
class Link : public BasicLink<link::platform::Clock>
{
public:
  Link(double bpm)
    : BasicLink(bpm)
  {
  }
};
```

Reply:
Done.